### DISCUSSION TITLE
Meta-calss Attributes should initialize to None

### DISCUSSION THREAD

**Discussion by fgagnonseed** [textX/textX#335] [Q&A]
It seems like the attributes of the meta-class are initialized to None for non-terminals but to some default values for terminals.  Consider the grammar below.

```
from textx import metamodel_from_str
grammar = """
  constant :
       float=FLOAT 
     | integer=INT 
     | true='true' 
     | false='false'
     | string=STRING
     | string2=my_literal
  ; 
my_literal : /\"(.*?)\"/ ;  
"""

parser = metamodel_from_str(grammar)
ast = parser.model_from_str("true")
print("ast.float [%s]"%  str(ast.float))
print("ast.integer [%s]"%str(ast.integer))
print("ast.true [%s]"%   str(ast.true))
print("ast.false [%s]"%  str(ast.false))
print("ast.string [%s]"% str(ast.string))
print("ast.string2[%s]"% str(ast.string2))
```

producing this output:
```
ast.float [0.0]
ast.integer [0]
ast.true [true]
ast.false []
ast.string []
ast.string2[None]
```

You can see that integers initialize to 0, floats to 0.0 and other strings to the empty string ('') while the none-terminal (string2) initializes to None.  I would argue that they should be consistent and all initialize to None unless they parsed.  Otherwise there is no reliable way to know which rule fired.  What if the user entered 0.0 or the empty string?  

Using the boolean assignment (?=) gives me the way to know which rule fired, but I can't retrieve the value (int/float/string/...) that was entered.

**Comment by igordejanovic** [comment]
That is expected behavior documented [here](http://www.igordejanovic.net/textX/stable/metamodel/#auto-initialization-of-the-attributes). You can change it with `auto_init_attributes` param during meta-model creation.

**Reply by mfrigerio17** [reply]
Hi, updated reference to the [right section of the manual](https://textx.github.io/textX/metamodel.html#auto-initialization-of-the-attributes)
